src/account_id_ref.rs

```diff
@@ -43,13 +43,18 @@ pub enum AccountType {
     NearImplicitAccount,
     /// An account which address starts with '0x', followed by 40 hex characters.
     EthImplicitAccount,
+    /// An account which address starts with '0s', followed by 40 hex characters.
+    ///
+    /// See [NEP-616](https://github.com/near/NEPs/pull/616/) for more details.
+    NearDeterministicAccount,
 }
 
 impl AccountType {
     pub fn is_implicit(&self) -> bool {
         match &self {
             Self::NearImplicitAccount => true,
             Self::EthImplicitAccount => true,
+            Self::NearDeterministicAccount => true,
```

Comment:
This might be a breaking change for existing contracts, such as [here](https://github.com/near/intents/blob/7ca45eda8ced3617af0678597851f8fd30667a18/defuse/src/contract/accounts/account/mod.rs#L58-L60): other devs (including myself) might think that `.is_implicit()` should return true iff this is an "EOA" account id that was derived from the public key.
```suggestion
            Self::NearDeterministicAccount => false,
```

Reply:
Hm, that's tricky. Wouldn't it equally be breaking for anybody assuming that `!is_implicit()` means it's a named account?

From my point of view, deterministic accounts have an implicit ID rather than an explicitly set name, so it makes more sense to say this is implicit. Nearcore code also treats this as an implicit account with regards to account creation, as you can send tokens to it before explicitly creating it with `CreateAccount`. I don't really see an argument as to why this would be an explicitly named account.

If we go with a 2.0.0 release anyway, we can mention this change in the changelog as breaking, which ever way we decide. But we must first align on what is the correct behaviour.

@mitinarseny what do you think?